### PR TITLE
Apply safe-args plugin

### LINKED ISSUE #72: Add safeargs for Navigation
https://developer.android.com/topic/libraries/architecture/navigation/navigation-implementing#Safe-args

### PR DESCRIPTION
**Pull request by andhie**
Fix #72

### CODE REVIEW

File: app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt [hunk 1]
```diff
@@ -48,9 +46,10 @@ class PlantAdapter : ListAdapter<Plant, PlantAdapter.ViewHolder>(PlantDiffCallba
     }
 
     private fun createOnClickListener(plantId: String): View.OnClickListener {
-        val bundle = bundleOf(PlantDetailFragment.ARG_ITEM_ID to plantId)
-        return Navigation.createNavigateOnClickListener(
-                R.id.action_plant_list_fragment_to_plant_detail_fragment, bundle)
+        return View.OnClickListener {
```
Comment (XinyueZ) [inline app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt]: make simple for reading 😄(I know they are the same thing, however, two times `View.OnClickListener`)  near position).
-> 
`fun foo() = View.OnClickListener{}`

Comment (andhie) [reply app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt]: actually i wanted to remove this click listener. didnt like having a click listener inside adapter. i havent fully think it through yet. lemme hear your thoughts on this.

1. define an interface `OnPlantClickedListener(View view, Plant plant)`. the view can be used for transition animation later.
2. implement in `Fragment` with the Navigation logic
3. In the XML of the list item, at the root will be `android:onClick="@{v -> listener.onPlantClicked(v, plant)}"`
4. When creating the Adapter, pass in the `OnPlantClickedListener` or create `setOnPlantClickedListener` method in adapter

Comment (XinyueZ) [reply app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantAdapter.kt]: @andhie forget it.  :( I have closed my PR by myself, there's really guided line for expression and function body . https://github.com/googlesamples/android-sunflower/pull/107

**Comment by andhie** [comment]
Done

**Comment by tiembo** [comment]
Thanks!

**Pull request merged**